src/ui/src/components/workflows/WorkflowsWorkflow.vue

```diff
@@ -209,55 +235,69 @@ function handleNodeDrag(ev: DragEvent, componentId: Component["id"]) {
 	if (x < 0 || y < 0) return;
 
 	const component = wf.getComponentById(componentId);
-	component.x = x - clickOffset.x;
-	component.y = y - clickOffset.y;
-	refreshArrows();
+
+	const newX = x - clickOffset.x;
+	const newY = y - clickOffset.y;
+
+	if (component.x == newX && component.y == newY) return;
+
+	component.x = newX;
+	component.y = newY;
+
+	setTimeout(() => {
+		// Debouncing
+		if (component.x !== newX) return;
+		if (component.y !== newY) return;
+		changeCoordinates(componentId, newX, newY);
+	}, 200);
```

Comment:
Yes that was the other alternative but I wanted to avoid having a single timer `id` for all components. So I'd have needed `Record<Component["id"], ReturnType<typeof setTimeout>>` and I wanted to avoid.

However, in the example you declare `id` inside the function. If it works that'd be great, but how would that behave? Wouldn't future calls get a completely new `id` since it's scoped to the function?